server/util/claims/claims.go

```diff
@@ -174,18 +174,23 @@ func ClaimsFromSubID(ctx context.Context, env environment.Env, subID string) (*C
 func userClaims(u *tables.User, effectiveGroup string) *Claims {
 	allowedGroups := make([]string, 0, len(u.Groups))
 	groupMemberships := make([]*interfaces.GroupMembership, 0, len(u.Groups))
+	cacheEncryptionEnabled := false
 	for _, g := range u.Groups {
 		allowedGroups = append(allowedGroups, g.Group.GroupID)
 		groupMemberships = append(groupMemberships, &interfaces.GroupMembership{
 			GroupID: g.Group.GroupID,
 			Role:    role.Role(g.Role),
 		})
+		if g.Group.GroupID == effectiveGroup {
+			cacheEncryptionEnabled = g.Group.CacheEncryptionEnabled
```

Comment:
You can save this for a separate PR, but we should also fix the value during impersonation. ClaimsFromSubID creates a synthetic Group entry that does not have any of the group attributes filled in apart from the GroupID